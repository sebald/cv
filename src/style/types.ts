import { Interpolation, ThemeProps } from 'styled-components';
import { theme } from './theme';

export type Theme = typeof theme;

export type ThemedInterpolation = Interpolation<ThemeProps<Theme>>;
