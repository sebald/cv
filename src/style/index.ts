export { default as theme, Theme } from './theme';

export * from './mixins';
export * from './styled-components';
export * from './utils';
