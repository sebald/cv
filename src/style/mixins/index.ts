export * from './animation';
export * from './breakpoints';
export * from './color';
export * from './space';
export * from './typography';
export * from './width';

export * from './types';
