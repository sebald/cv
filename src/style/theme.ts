export const theme = {
  baseline: 12,
  htmlFontSize: 16,
};
