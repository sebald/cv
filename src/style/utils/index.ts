export * from './classnames';
export * from './constants';
export * from './createMixinComponent';
export * from './unit';
export * from './typography';
