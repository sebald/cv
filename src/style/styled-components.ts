import { CSSProperties } from 'react';
import * as styledComponents from 'styled-components';
import {
  ThemedStyledComponentsModule,
  ThemedStyledFunction,
} from 'styled-components';

import { Theme } from './theme';

type StyledProps<T extends string> = Pick<CSSProperties, T>;

/**
 * Make styled-components work with props.
 * More info: https://github.com/styled-components/styled-components/issues/630#issuecomment-317172454
 */
type TTag = keyof JSX.IntrinsicElements;
function suit<U>(tag: TTag) {
  return styled[tag] as ThemedStyledFunction<
    JSX.IntrinsicElements[TTag] & U,
    Theme
  >;
}

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider,
} = styledComponents as ThemedStyledComponentsModule<Theme>;

export {
  styled,
  css,
  injectGlobal,
  keyframes,
  suit,
  StyledProps,
  ThemeProvider,
  TTag,
};
