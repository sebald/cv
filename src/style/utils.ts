import { theme } from './theme';

const { htmlFontSize } = theme;

export const px = (value: number) => `${value}px`;

export const pxToRem = (pixel: number, base = htmlFontSize) =>
  `${parseFloat((pixel / base).toFixed(5))}rem`;

export const remToPx = (rem: string, base = htmlFontSize) =>
  parseFloat(rem) * base;

export const ensurePx = (value: number | string, base = htmlFontSize) =>
  typeof value === 'string' ? remToPx(value, base) : value;
