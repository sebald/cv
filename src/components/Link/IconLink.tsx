import { StyledComponentClass } from 'styled-components';
import { color, Theme } from '../../style';

import { Icon } from '../Icon';
import { Link } from './Link';

const IconWithLink: StyledComponentClass<
  React.ClassAttributes<HTMLAnchorElement> &
    React.AnchorHTMLAttributes<HTMLAnchorElement>,
  Theme
> = Icon.withComponent(Link);

export const IconLink = IconWithLink.extend`
  color: ${color(palette => palette.grey['300'])};

  &:hover {
    color: ${color(palette => palette.grey['600'])};
  }
`;
