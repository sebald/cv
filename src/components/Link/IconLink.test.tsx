import { shallow } from 'enzyme';
import * as React from 'react';

import { theme } from '../../style';
import { IconLink } from './IconLink';

test('render correct', () => {
  const wrapper = shallow(<IconLink theme={theme} />);
  expect(wrapper).toMatchSnapshot();
});
