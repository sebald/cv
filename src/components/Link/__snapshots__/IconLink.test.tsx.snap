// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correct 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  width: 2rem;
  color: #e0e0e0;
}

.c0:hover {
  color: #757575;
}

<styled.a
  className="c0"
  theme={
    Object {
      "animation": Object {
        "fadeInUp": "eqHJzO",
      },
      "background": Object {
        "body": "#fff",
      },
      "breakpoints": Object {
        "large": 1280,
        "medium": 1024,
        "small": 768,
      },
      "layout": Object {
        "content": 800,
        "sidebar": 350,
      },
      "palette": Object {
        "black": "#000",
        "grey": Object {
          "100": "#f5f5f5",
          "200": "#eeeeee",
          "300": "#e0e0e0",
          "400": "#bdbdbd",
          "50": "#fafafa",
          "500": "#9e9e9e",
          "600": "#757575",
          "700": "#616161",
          "800": "#424242",
          "900": "#212121",
        },
        "white": "#fff",
      },
      "space": Object {
        "baseline": 16,
      },
      "typography": Object {
        "body": Object {
          "color": "#757575",
          "fontFamily": "'Raleway', sans-serif;",
          "fontSize": "1rem",
          "fontWeight": 300,
          "lineHeight": "2rem",
          "marginBottom": "0.375rem",
          "marginTop": 0,
          "paddingTop": "0.625rem",
        },
        "caption": Object {
          "color": "#bdbdbd",
          "fontFamily": "'Montserrat', sans-serif;",
          "fontSize": "1rem",
          "fontWeight": 300,
          "lineHeight": "1rem",
          "marginBottom": "0.875rem",
          "marginTop": 0,
          "paddingTop": "0.125rem",
        },
        "display": Object {
          "color": "#212121",
          "fontFamily": "'Montserrat', sans-serif;",
          "fontSize": "3.375rem",
          "fontWeight": 300,
          "letterSpacing": "-1px",
          "lineHeight": "4rem",
          "marginBottom": "0.25rem",
          "marginTop": 0,
          "paddingTop": "0.75rem",
        },
        "fontFamily": Object {
          "body": "'Raleway', sans-serif;",
          "heading": "'Montserrat', sans-serif;",
        },
        "heading": Object {
          "color": "#212121",
          "fontFamily": "'Montserrat', sans-serif;",
          "fontSize": "2.25rem",
          "fontWeight": 400,
          "lineHeight": "4rem",
          "marginBottom": "0.8125rem",
          "marginTop": 0,
          "paddingTop": "0.1875rem",
          "textTransform": "uppercase",
        },
        "htmlFontSize": 16,
        "subheading": Object {
          "color": "#212121",
          "fontFamily": "'Montserrat', sans-serif;",
          "fontSize": "1.5rem",
          "fontWeight": 300,
          "lineHeight": "2rem",
          "marginBottom": "0.5625rem",
          "marginTop": 0,
          "paddingTop": "0.4375rem",
          "textTransform": "uppercase",
        },
      },
    }
  }
/>
`;
